ivy/stateful/module.py

```diff
@@ -144,6 +144,9 @@ def __init__(
         self._target = None
         self._lazy_compiled = False
         self._dynamic_backend = dynamic_backend
+        if hasattr(self, "_create_buffers"):
```

Comment:
I'm not sure why we need to have a tracker for buffers, I think `module.create_buffer` could just add something to the `self._buffers` and then we can have a read-only property `buffers` to get the value. What I mean is we don't need to explicitly allow a overridable `_create_buffers` function analogous to `_create_variables`, as all custom usages should always flow through `self.create_buffer` which could be called anywhere in the module (in the `__init__`, `_build`, `__call__`, `_forward`, etc.) and whenever we come across a call like that, we add it to the `self._buffers` dict. 
I think this will simplify the approach quite a bit, happy to know what you think. Thanks @RickSanchezStoic 😄

Reply:
The `_create_buffers` is just analogous to `_create_variables` , where users can define variables specifically meant to be buffer variables, unlike registering it on spot like torch (support for which is also added) , thought would be a good addon

Reply:
I think given that the other frameworks follow the approach of creating buffers on spot ([torch](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.register_buffer) and [haiku](https://dm-haiku.readthedocs.io/en/latest/api.html#get-state)), I think our approach should also mirror theirs. I'm not sure if we'd add a lot of value by providing an overridable method for users to explicitly register buffers, the on-spot way would provide quite a lot of flexibility.
Just as a side note, we'd of course need to also store the buffers created in child modules in the buffer_dict or state_dict of the module where those child modules are initialized, just like we've done with variables.
Happy to know what you think, thanks @RickSanchezStoic 😄